runtime/rasdump/javadump.cpp

```diff
@@ -613,35 +621,123 @@ JavaCoreDumpWriter::writeTitleSection(void)
 		writeEventDrivenTitle();
 	}
 
-	/* Write the date and time that the dump was generated */
-	U_64 now = j9time_current_time_millis();
-	RasDumpGlobalStorage* dump_storage;
-	struct J9StringTokens* stringTokens;
+	/* write the date and time that the dump was generated */
+	char timeStamp[_MaximumTimeStampLength + 1];
+	int64_t now = j9time_current_time_millis();
+	bool zoneAvailable = false;
+	int32_t zoneSecondsEast = 0;
+	const char *zoneName = NULL;
 
-	dump_storage = (RasDumpGlobalStorage*)_VirtualMachine->j9rasdumpGlobalStorage;
+	strncpy(timeStamp, "(unavailable)", _MaximumTimeStampLength);
 
-	/* lock access to the tokens */
-	omrthread_monitor_enter(dump_storage->dumpLabelTokensMutex);
-	stringTokens = (struct J9StringTokens*)dump_storage->dumpLabelTokens;
-	j9str_set_time_tokens(stringTokens, now);
-	/* release access to the tokens */
-	omrthread_monitor_exit(dump_storage->dumpLabelTokensMutex);
+#if defined(WIN32)
+	time_t timeNow = time(NULL);
+	struct tm utc;
+
+	if (0 == gmtime_s(&utc, &timeNow)) {
+		strftime(timeStamp, _MaximumTimeStampLength, TIMESTAMP_FORMAT, &utc);
+		timeStamp[_MaximumTimeStampLength] = '\0';
+		/* until a reliable mechanism is found, don't report timezone */
+	}
+#elif defined(J9ZOS390) /* defined(WIN32) */
+	time64_t timeNow = time64(NULL);
+	struct tm utc;
+	struct tm local;
+
+	if (NULL != gmtime64_r(&timeNow, &utc)) {
+		strftime(timeStamp, _MaximumTimeStampLength, TIMESTAMP_FORMAT, &utc);
+		timeStamp[_MaximumTimeStampLength] = '\0';
+
+		if (NULL != localtime64_r(&timeNow, &local)) {
+			zoneAvailable = true;
+			zoneSecondsEast = (int32_t)difftime64(timeNow, mktime64(&utc));
+			if (0 == local.tm_isdst) {
+				zoneName = tzname[0];
+			} else if (local.tm_isdst > 0) {
+				zoneName = tzname[1];
+				/* compensate for DST because difftime64() doesn't appear to do so */
+				zoneSecondsEast += 60 * 60;
+			}
+		}
+	}
+#else /* defined(WIN32) */
+	time_t timeNow = time(NULL);
+	struct tm utc;
+	struct tm local;
+
+	if (NULL != gmtime_r(&timeNow, &utc)) {
+		strftime(timeStamp, _MaximumTimeStampLength, TIMESTAMP_FORMAT, &utc);
+		timeStamp[_MaximumTimeStampLength] = '\0';
+
+		if (NULL != localtime_r(&timeNow, &local)) {
+			zoneAvailable = true;
+			zoneSecondsEast = (int32_t)difftime(timeNow, mktime(&utc));
+			if (0 == local.tm_isdst) {
+				zoneName = tzname[0];
+			} else if (local.tm_isdst > 0) {
+				zoneName = tzname[1];
+				/* compensate for DST because difftime() doesn't appear to do so */
+				zoneSecondsEast += 60 * 60;
+			}
+		}
+	}
+#endif  /* defined(WIN32) */
 
-	char timeStamp[_MaximumTimeStampLength];
+	_OutputStream.writeCharacters("1TIDATETIMEUTC Date: ");
+	_OutputStream.writeCharacters(timeStamp);
+	_OutputStream.writeInteger(now % 1000, ":%03d"); /* add the milliseconds */
```

Comment:
It is possible for the seconds to roll over between getting j9time_current_time_millis() and gmtime_r(). Maybe the millis should be left off here.

Reply:
j9time_current_time_millis() is in UTC, another solution is to add a way to print this in UTC rather than fetching the time again.

Reply:
It's actually the `time(NULL)` call that's relevant, but I take your point. I will remove the milliseconds from the UTC line.

Reply:
I'll look into creating `omrstr_ftime_utc()` which would be like `omrstr_ftime()` except that it would not convert to local time.